src/foundry/common/data/data.mjs/chatMessageData.d.ts

```diff
@@ -0,0 +1,207 @@
+import { FieldReturnType, PropertiesToSource } from '../../../../types/helperTypes';
+import { DocumentData } from '../../abstract/module.mjs';
+import { CONST, documents } from '../../module.mjs';
+import * as fields from '../fields.mjs';
+import { ChatSpeakerData, ChatSpeakerDataConstructorData } from './chatSpeakerData';
+
+interface ChatMessageDataSchema extends DocumentSchema {
+  _id: typeof fields.DOCUMENT_ID;
+  type: DocumentField<CONST.ChatMessageType> & {
+    type: typeof Number;
+    required: true;
+    default: typeof CONST.CHAT_MESSAGE_TYPES.OTHER;
+    validate: typeof _validateChatMessageType;
+    validationError: 'The provided ChatMessage type must be in CONST.CHAT_MESSAGE_TYPES';
+  };
+  user: fields.ForeignDocumentField<{ type: documents.BaseUser; required: true }>;
+  timestamp: FieldReturnType<typeof fields.TIMESTAMP_FIELD, { required: true }>;
+  flavor: typeof fields.STRING_FIELD;
+  content: typeof fields.BLANK_STRING;
+  speaker: DocumentField<typeof ChatSpeakerData> & { required: true; default: {} };
```

Comment:
The `type` property needs to be set explicitly. This is not covered by the generic of `DocumentField`.

Reply:
I guess it should not be `typeof ChatSpeakerData` also

Reply:
I guess it should not be `typeof ChatSpeakerData` but `ChatSpeakerData`